File: generated-code/example/go/TcpReadWrite/model/enumeration.go
package model

import "io"
import . "trans_gen_test/stream"

// Example enumeration
type Enumeration int32

const (
    // First option
    EnumerationValueOne Enumeration = 0
    // Second option
    EnumerationValueTwo Enumeration = 1
)

// Read Enumeration from reader
func ReadEnumeration(reader io.Reader) Enumeration {
    switch ReadInt32(reader) {
    case 0:
        return EnumerationValueOne
    case 1:
        return EnumerationValueTwo
    }
    panic("Unexpected tag value")
}

// Get string representation of Enumeration
func EnumerationToString(enumeration Enumeration) string {
    switch enumeration {
    case EnumerationValueOne:
        return "ValueOne"
    case EnumerationValueTwo:
        return "ValueTwo"
    }
    panic("Impossible happened")
}

File: generated-code/example/csharp/TcpReadWrite/Model/Enumeration.cs
namespace TransGenTest.Model
{
    /// <summary>
    /// Example enumeration
    /// </summary>
    public enum Enumeration
    {
        /// <summary>
        /// First option
        /// </summary>
        ValueOne = 0,
        /// <summary>
        /// Second option
        /// </summary>
        ValueTwo = 1,
    }

    public static class EnumerationHelper {
        /// <summary> Read Enumeration from reader </summary>
        public static Enumeration ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return Enumeration.ValueOne;
                case 1:
                    return Enumeration.ValueTwo;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}